api/appStore/AppStoreRouter.go

```diff
@@ -29,27 +29,34 @@ type AppStoreRouter interface {
 }
 
 type AppStoreRouterImpl struct {
-	deployRestHandler        InstalledAppRestHandler
-	appStoreValuesRouter     appStoreValues.AppStoreValuesRouter
-	appStoreDiscoverRouter   appStoreDiscover.AppStoreDiscoverRouter
-	appStoreDeploymentRouter appStoreDeployment.AppStoreDeploymentRouter
+	deployRestHandler                 InstalledAppRestHandler
+	appStoreValuesRouter              appStoreValues.AppStoreValuesRouter
+	appStoreDiscoverRouter            appStoreDiscover.AppStoreDiscoverRouter
+	appStoreDeploymentRouter          appStoreDeployment.AppStoreDeploymentRouter
+	appStoreStatusTimelineRestHandler AppStoreStatusTimelineRestHandler
 }
 
 func NewAppStoreRouterImpl(restHandler InstalledAppRestHandler,
 	appStoreValuesRouter appStoreValues.AppStoreValuesRouter, appStoreDiscoverRouter appStoreDiscover.AppStoreDiscoverRouter,
-	appStoreDeploymentRouter appStoreDeployment.AppStoreDeploymentRouter) *AppStoreRouterImpl {
+	appStoreDeploymentRouter appStoreDeployment.AppStoreDeploymentRouter,
+	appStoreStatusTimelineRestHandler AppStoreStatusTimelineRestHandler) *AppStoreRouterImpl {
 	return &AppStoreRouterImpl{
-		deployRestHandler:        restHandler,
-		appStoreValuesRouter:     appStoreValuesRouter,
-		appStoreDiscoverRouter:   appStoreDiscoverRouter,
-		appStoreDeploymentRouter: appStoreDeploymentRouter,
+		deployRestHandler:                 restHandler,
+		appStoreValuesRouter:              appStoreValuesRouter,
+		appStoreDiscoverRouter:            appStoreDiscoverRouter,
+		appStoreDeploymentRouter:          appStoreDeploymentRouter,
+		appStoreStatusTimelineRestHandler: appStoreStatusTimelineRestHandler,
 	}
 }
 
 func (router AppStoreRouterImpl) Init(configRouter *mux.Router) {
 	// deployment router starts
 	appStoreDeploymentSubRouter := configRouter.PathPrefix("/deployment").Subrouter()
 	router.appStoreDeploymentRouter.Init(appStoreDeploymentSubRouter)
+
+	configRouter.Path("/deployment-status/timeline/{installedAppId}/{envId}").
```

Comment:
installed app id is already mapped with environment id

Reply:
I'm directly getting installedAppVersion from these two ids, hence saving one db call in which I would have needed to fetch installed apps from db then  get env id from installed app object then pass env id to function